app/assets/javascripts/controllers/deploy_groups_ctrl.js

```diff
@@ -1,15 +1,46 @@
 samson.controller('DeployGroupsCtrl', function($scope, $http, $location, $window) {
+  var MAX_PAGES = 20;
 
   function init() {
-    $http.get($location.path() + '.json').success(function(result) {
-      result.deploys.forEach(function(deploy) {
-        deploy.start = new Date(deploy.started_at || deploy.created_at);
+    getMoreDeploys(1);
+  }
+
+  function getMoreDeploys(page) {
+    if (page < MAX_PAGES) {
+      $http.get($location.path() + '.json?page=' + page).success(function(result) {
+        if (result.deploys.length > 0) {
+          result.deploys.forEach(function(deploy) {
+            deploy.start = new Date(deploy.started_at || deploy.created_at);
+          });
+
+          $scope.items.add(result.deploys);
+          $scope.deployIds = $scope.deployIds.concat(_.pluck(result.deploys, 'id'));
+
+          adjustVisibleItems(result.deploys);
+          getMoreDeploys(page + 1);
+        }
+      });
+    }
+  }
+
+  function adjustVisibleItems() {
+    var visibleItems = $scope.timeline.getVisibleItems().length,
+        MIN_VIS_ITEMS = 10,
+        MAX_VIS_ITEMS = 20;
+
+    if (visibleItems > MAX_VIS_ITEMS) {
+      $scope.timeline.setOptions({
+        start: $scope.items.get($scope.deployIds[MAX_VIS_ITEMS-1]).start
+      });
+    } else if (visibleItems <= MIN_VIS_ITEMS && $scope.items.length > MIN_VIS_ITEMS) {
+      $scope.timeline.setOptions({
+        start: $scope.items.get($scope.deployIds[MIN_VIS_ITEMS-1]).start
```

Comment:
this appears to be the default behaviour unless I'm misreading `start` [here](http://visjs.org/docs/timeline.html#Configuration_Options)